commons/all/src/main/java/org/infinispan/commons/api/query/Query.java

```diff
@@ -23,6 +27,22 @@ public interface Query<T> extends Iterable<T> {
     */
    List<T> list();
 
+   /**
+    * Reactive based query for {@link #list()}. Will stream data as requested. The query will poll data
+    * from the underlying engine in <b>batchSize</b> blocks of items to satisfy the subscription
+    * <p>
+    * If {@link #startOffset(long)} or {@link #maxResults(int)} will affect the query by either not
+    * returning the first <b>startOffset</b> number of matches and will not return more than <b>maxResults</b>.
+    * <p>
+    * Note: due to current API limitations it is not safe to use this Query instance while there is an active
+    * subscription to the returned Publisher and it is only safe to use one subscription at a time.
+    * @return a Publisher that when subscribed will query
+    */
+   @Experimental
+   default Publisher<T> listAsync(int batchSize) {
```

Comment:
nitpicking: I think the method should be named `publish()` or something similar. It doesn't return a list :)

Reply:
Sure, sounds good 👍